src/__tests__/encoder.ts

```diff
@@ -0,0 +1,17 @@
+import { Hex } from 'viem';
+
+import { Builder, IBuilder } from '@/__tests__/builder';
+
+export interface IEncoder<T, E = Hex> extends IBuilder<T> {
+  encode(): E;
+}
+
+export class Encoder<T, E = Hex> extends Builder<T> implements IEncoder<T, E> {
+  encode(): E {
+    throw new Error('Method not implemented.');
+  }
```

Comment:
`encode` does seem to hint that this might be better suited for an abstract method.

Since the base `Builder` class already has the required properties minus the encoding functionality probably the `Encoder` class can be removed altogether and current implementations of this subclass would implement the `IEncoder` interface and extend the `Builder` class. What do you think?

Reply:
As per discussion, we will iterate on this afterwards.